ARMC5 or uArm not in Mbed OS 6

iriark01 opened the pull request:
I've made some assumptions here. For example, I assumed ARMC6 can work wherever we say ARMC5. Please check each line very carefully to make sure I'm not misleading users.

iriark01 committed:
ARMC5 or uArm not in Mbed OS 6

iriark01 committed:
That was confusing

iriark01 commented:
I think there should also be an update to the Bare Metal content, because of the uARM change, but that's beyond my ability to edit at this point.

hugueskamba commented on docs/api/io/alarm-tutorial.md:
```diff
@@ -19,7 +19,7 @@ If using the Mbed Online Compiler, use the **Import into Mbed IDE** button below
 ### Compile
 
-Invoke `mbed compile`, and specify the name of your platform and your favorite toolchain (`GCC_ARM`, `ARM`, `IAR`). For example, for the ARM Compiler 5:
+Invoke `mbed compile`, and specify the name of your platform and your favorite toolchain (`GCC_ARM`, `ARM`, `IAR`). For example, for the ARM Compiler:
```
```suggestion
Invoke `mbed compile`, and specify the name of your platform and your favorite toolchain (`GCC_ARM`, `ARM`, `IAR`). For example, for the ARM Compiler and K64F platform:
```
?

hugueskamba commented on docs/program-setup/mbed_targets.md:
```diff
@@ -194,7 +194,9 @@ Enabling `is_disk_virtual` adds delay after flashing firmware binary to make sur
 ### supported_toolchains
 
-The `supported_toolchains` property is the list of toolchains that support a target. The allowed values for `supported_toolchains` are `ARM`, `uARM`, `ARMC5`, `ARMC6`, `GCC_ARM` and `IAR`.
+The `supported_toolchains` property is the list of toolchains that support a target. The allowed values for `supported_toolchains` are `ARM`, `ARMC5`, `UARM`, `ARMC6`, `GCC_ARM` and `IAR`.
```
We should probably also mention `supported_c_libs` as it is also related to toolchain support.

iriark01 replied:
Is that related to the Bare Metal question I had?

hugueskamba replied:
Hmm I am not sure which question you mean.
`supported_c_libs` defines the C libraries that can be used for each `supported_toolchains`. When the bare metal profile is used, we recommend to use `small` C libraries if available for that toolchain. See Mbed 2 target porting guide for more details.

iriark01 replied:
The uArm question. To my understanding, from 6 we don't support uArm.

hugueskamba commented on docs/program-setup/mbed_targets.md:
```diff
@@ -194,7 +194,9 @@ Enabling `is_disk_virtual` adds delay after flashing firmware binary to make sur
+
+<span class="notes">**Note**: Mbed OS 6 does not support Arm Compiler 5 or uARM.</span>
 
 When using `ARM`, `ARMC5`, `ARMC6` for `supported_toolchains`, please note:
```
Should we not get rid of every mentions of `ARMC5` except to say it is no longer supported?

iriark01 replied:
I did that at first.  But if you look at the list of values, is it still there? That may confuse people.

hugueskamba replied:
Sorry which list of values? Which section (or line number)?

iriark01 replied:
The list of values that can go in supported_toolchains. Is there anywhere you can query that list to see all the values, and will that query return uArm and ARMC5?

hugueskamba commented on docs/quick-start/cli_code.md:
```diff
@@ -11,7 +11,7 @@
 1. Compile and program your board:
 
-   Invoke `mbed compile`, and specify the name of your platform and your installed toolchain (`GCC_ARM`, `ARM`, `IAR`). For example, for the K64F platform and Arm Compiler 5 toolchain:
+   Invoke `mbed compile`, and specify the name of your platform and your installed toolchain (`GCC_ARM`, `ARM`, `IAR`). For example, for the K64F platform and Arm Compiler toolchain:
```
```suggestion
   Invoke `mbed compile`, and specify the name of your platform and your installed toolchain (`GCC_ARM`, `ARM`, `IAR`). For example, for the K64F platform and ARM toolchain:
```

hugueskamba commented on docs/tools/debug/memap.md:
```diff
@@ -72,4 +72,4 @@ There are other entries that require a bit of clarification:
 ## Current support
 
-We have tested memap on Windows 7, Linux and macOS. The `GCC_ARM` (GNU Arm Embedded Toolchain), Arm Compiler 5, Arm Compiler 6 and IAR toolchains generate memory map files.
+We have tested memap on Windows 7, Linux and macOS. The `GCC_ARM` (GNU Arm Embedded Toolchain), Arm Compiler 6 and IAR toolchains generate memory map files.
```
Which Linux distro? Which OS X version?

iriark01 replied:
I don't know. I'll try to find out who did the testing.

hugueskamba replied:
Thanks as it may not work with latest versions of OS X or with another Linux distribution.

iriark01 committed:
Update docs/api/io/alarm-tutorial.md

Co-Authored-By: Hugues Kamba <hugues.kamba@arm.com>

iriark01 committed:
Update docs/quick-start/cli_code.md

Co-Authored-By: Hugues Kamba <hugues.kamba@arm.com>

iriark01 committed:
Removing deprecated API

It's breaking the build

iriark01 committed:
c_small_libs update

But there's a whole "if/then" section there I will need help with

iriark01 commented on docs/program-setup/bare_metal/c_small_libs.md:
````diff
@@ -56,7 +56,3 @@ After you have completed the steps above, add `small` to the `supported_c_libs`
     "iar": ["std"]
 }
 ```
````
@hugueskamba  there's a whole explanation here about uARM in a sort of if/then method. I suppose it needs to be updated.

hugueskamba replied:
uARM support has been removed so there should be anything about it.

iriark01 replied:
Checked with Evelyne; she said it's still correct for bare metal

iriark01 commented:
@hugueskamba could you help with this?

hugueskamba commented on docs/api/io/alarm-tutorial.md:
```diff
@@ -19,7 +19,7 @@ If using the Mbed Online Compiler, use the **Import into Mbed IDE** button below
 ### Compile
 
-Invoke `mbed compile`, and specify the name of your platform and your favorite toolchain (`GCC_ARM`, `ARM`, `IAR`). For example, for the ARM Compiler 5:
+Invoke `mbed compile`, and specify the name of your platform and your favorite toolchain (`GCC_ARM`, `ARM`, `IAR`). For example, for the ARM Compiler and K64F platform:
```
You may want to remove the parentheses listing the toolchains as my understanding is that support for IAR will be dropped soon. Readers should refer to another page to know which toolchains are supported as there should be a single source of truth.Maybe link to the a page that is the source of truth for supported toolchains? (../tools/index.html#compiler-versions) ?

hugueskamba commented on docs/program-setup/mbed_targets.md:
```diff
@@ -194,7 +194,9 @@ Enabling `is_disk_virtual` adds delay after flashing firmware binary to make sur
-The `supported_toolchains` property is the list of toolchains that support a target. The allowed values for `supported_toolchains` are `ARM`, `uARM`, `ARMC5`, `ARMC6`, `GCC_ARM` and `IAR`.
+The `supported_toolchains` property is the list of toolchains that support a target. The allowed values for `supported_toolchains` are `ARM`, `ARMC5`, `UARM`, `ARMC6`, `GCC_ARM` and `IAR`.
+
+<span class="notes">**Note**: Mbed OS 6 does not support Arm Compiler 5 or uARM.</span>
```
```suggestion
<span class="notes">**Note**: Support for the Arm Compiler 5 and the uARM toolchain were dropped from Mbed OS 6.</span>
```
Something thing like that for making the text future proof after Mbed OS 6.

iriark01 replied:
"from Mbed OS 6" at the end of a sentence sounds like "from location", not "from version".

hugueskamba commented on docs/quick-start/cli_code.md:
```diff
@@ -11,7 +11,7 @@
 1. Compile and program your board:
 
-   Invoke `mbed compile`, and specify the name of your platform and your installed toolchain (`GCC_ARM`, `ARM`, `IAR`). For example, for the K64F platform and Arm Compiler 5 toolchain:
+   Invoke `mbed compile`, and specify the name of your platform and your installed toolchain (`GCC_ARM`, `ARM`, `IAR`). For example, for the K64F platform and ARM toolchain:
```
Same comment about as above regarding listing the toolchains supported.

iriark01 replied:
Does it actually still allow ARM? Is ARM now equal to ARM6?

Because when I look at where the single source of truth may be, the mapping between Mbed CLI's supported versions and these parameter values isn't obvious. 

```
- Arm Compiler 6.13 (default ARM toolchain).
  - A paid version is available as [Arm Compiler 6.13 Professional](https://developer.arm.com/products/software-development-tools/compilers/arm-compiler/downloads/version-6).
  - A paid version is also included in [Keil MDK 5.29](http://www2.keil.com/mdk5/529).
- [GNU Arm Embedded version 9 (9-2019-q4-major)](https://developer.arm.com/open-source/gnu-toolchain/gnu-rm/downloads).
- [IAR Embedded Workbench 8.32.1](https://iar.com/mbed).
```

hugueskamba commented on docs/tools/CLI/cli-compile.md:
```diff
@@ -12,12 +12,12 @@ The valid toolchain values are:
 | `IAR` | IAR EWARM Compiler |
 | `GCC_ARM` | GNU Arm Embedded Compiler (GCC) |
 
+<span class="notes">**Note**: Mbed OS 6 does not support Arm Compiler 5 or uARM; using `ARM` will default to Arm Compiler 6.</span>
```
Similar comment making the text future proof.

hugueskamba commented on docs/tools/CLI/cli-setup/cli-reqs.md:
````diff
@@ -65,15 +65,7 @@ pip install -U mbed-cli
-$ sudo apt-get update
-$ sudo apt-get install libc6:i386 libncurses5:i386 libstdc++6:i386
-```
+Download and install one of the following compilers: [GCC Arm, Arm Compiler 6 or IAR](../tools/index.html#compiler-versions).
````
It is not just the compilers that are installed. I think the text should say:
```suggestion
Install one of the [supported toolchains](../tools/index.html#compiler-versions).
```

hugueskamba commented on docs/tools/CLI/cli-test-debug.md:
```diff
@@ -9,7 +9,7 @@ There are two testing frameworks: Greentea and Icetea. Greentea provides tests d
 
 - `-m <MCU>`: to select a target for the compilation. If the `detect` or `auto` parameter is passed, then Mbed CLI will attempt to detect the connected target and compile against it.
-- `-t <TOOLCHAIN>`: to select a toolchain from those defined in `mbed_settings.py`, where `toolchain` can either be `ARM` (Arm Compiler 5), `GCC_ARM` (GNU Arm Embedded), or `IAR` (IAR Embedded Workbench for Arm).
+- `-t <TOOLCHAIN>`: to select a toolchain from those defined in `mbed_settings.py`, where `toolchain` can either be `ARM` (Arm Compiler 6), `GCC_ARM` (GNU Arm Embedded), or `IAR` (IAR Embedded Workbench for Arm).
```
```suggestion
- `-t <TOOLCHAIN>`: to select a toolchain from those defined in `mbed_settings.py`, where `toolchain` can either be `ARM` (Arm Compiler), `GCC_ARM` (GNU Arm Embedded), or `IAR` (IAR Embedded Workbench for Arm).
```
Just saw you don't keep updating this page when the version we use changes. There is a source of truth for the supported version of each toolchain (../tools/index.html#compiler-versions).

hugueskamba commented on docs/tools/toolchains/export_to_third_party.md:
```diff
@@ -6,16 +6,13 @@ If you'd like to develop on Arm Mbed OS with a third party tool, or migrate to o
 - IAR Systems.
 - Eclipse CDT ([C/C++ Development Tooling](https://www.eclipse.org/cdt/)) make (unmanaged) projects using:
     - GNU Arm Embedded Toolchain 6 (6-2017-q1-update).
-    - Arm Compiler 5.06 Update 6.
```
Do we know if it works with Arm Compiler 6? If it does, we should replace with Arm Compiler 6. We should check with DSG. They should know straightaway.

hugueskamba commented on docs/tools/toolchains/export_to_third_party.md:
```diff
@@ -97,17 +94,17 @@ make: *** [all] Error 2
 If you see such an error, remove the `-Wl,--wrap,_malloc_r`, `-Wl,--wrap,_free_r`, `-Wl,--wrap,_realloc_r` and `-Wl,--wrap,_calloc_r`from the `LD_FLAGS` variable in the Makefile.
 
-### Make and Eclipse (GNU Arm Embedded Toolchain, IAR Compiler, Arm Compiler 5)
+### Make and Eclipse (GNU Arm Embedded Toolchain, IAR Compiler, Arm Compiler 6)
```
```suggestion
### Make and Eclipse (GNU Arm Embedded Toolchain, IAR Compiler, Arm Compiler)
```

iriark01 replied:
Why remove the version indication here?

hugueskamba commented on docs/tools/toolchains/export_to_third_party.md:
```diff
@@ -97,17 +94,17 @@ make: *** [all] Error 2
+- Arm Compiler 6.
 
-<span class="notes">**Note:** Ensure that the compiler you are exporting to is accessible using your `PATH` environment variable because Makefile requires this. For example, when using an exported Makefile from `make_armc5`, the command `armcc` prints a help message about how to use Arm Compiler 5.</span>
+<span class="notes">**Note:** Ensure that the compiler you are exporting to is accessible using your `PATH` environment variable because Makefile requires this. For example, when using an exported Makefile from `make_armc6`, the command `armcc` prints a help message about how to use Arm Compiler 6.</span>
```
There is no `armcc` with Arm Compiler 6. It has been replaced with `armclang`.

iriark01 replied:
```suggestion
<span class="notes">**Note:** Ensure that the compiler you are exporting to is accessible using your `PATH` environment variable because Makefile requires this. For example, when using an exported Makefile from `make_armc6`, the command `armclang` prints a help message about how to use Arm Compiler 6.</span>
```

hugueskamba commented on docs/tools/tools_intro.md:
```diff
@@ -19,12 +19,12 @@ The Mbed Online Compiler is our in-house IDE and should be familiar to anyone wh
 
 | Architecture version | Compiler |
```
I would just get rid of the table. I do not see what useful information it gives and we are currently using the latest version of the toolchain.

hugueskamba commented on docs/tools/tools_intro.md:
```diff
@@ -19,12 +19,12 @@ The Mbed Online Compiler is our in-house IDE and should be familiar to anyone wh
 | Arm v8M       | `Arm Compiler 6.11` |
 
-<span class="notes">**Note**: Arm Compiler 6 is the default Arm Compiler version for Mbed OS development. All platforms are already compatible with it. Please do not use Arm Compiler 5 in any new development as it will be deprecated in the future.</span>
+<span class="notes">**Note**: Mbed OS 6 does not support Arm Compiler 5 or uARM.</span>
```
Same comment to make this line future proof.

iriark01 replied:
I'm sorry, I thought I left a question here. 

https://os.mbed.com/docs/mbed-os/development/build-tools/index.html

There is a list of supported compilers for Mbed CLI. That *seems* like a useful "single source of truth" (without the ARMC5 line, of course). So going with your suggestion, tidying up this list and making it more prominent so I can link everywhere else to it seems useful. 

Can you tie the names of the compilers in this list to their parameter names? For example, is there any use for `ARM` when now it can only mean `ARMC6`?

`ARM`, `ARMC5`, `UARM`, `ARMC6`, `GCC_ARM` and `IAR`

iriark01 committed:
A couple of updates

MarceloSalazar commented on docs/tools/CLI/cli-setup/cli-reqs.md:
```diff
@@ -110,7 +101,6 @@ You can set the location of the compiler with environment variables. Use the pat
 | --------- | --------- |
-| Arm Compiler 5 | `MBED_ARM_PATH` |
 | Arm Compiler 6 | `MBED_ARMC6_PATH` |
 | IAR EWARM Compiler | `MBED_IAR_PATH` |
```
Note IAR is not supported in Mbed OS 6.0, therefore everything related to IAR should be removed.

MarceloSalazar commented on docs/tools/CLI/cli-setup/cli-reqs.md:
```diff
@@ -98,7 +90,6 @@ Mbed CLI supports a setting for each toolchain path:
 | Toolchain | Setting name | Example binary location | Example path |
 | --------- | --------- | ---------| --------- |
-| Arm Compiler 5.06u6 | `ARM_PATH` | `C:\Program Files\ARM_Compiler_5.06u6\bin\armcc` | `C:\Program Files\ARM_Compiler_5.06u6` |
 | Arm Compiler 6.11 | `ARMC6_PATH` | `C:\Program Files\ARMCompiler6.11\bin\armclang` | `C:\Program Files\ARMCompiler6.11\bin` |
```
We should update the version of AC6 to ~Arm Compiler 6.13~, as it's the one supported and tested in CI and Mbed Studio.
EDIT: I'm keen on removing specific compiler versions of this file or use something generic such `6.xx` and document supported compiler versions in a single location, for example here `tools_intro.md`

iriark01 replied:
Yes, Hugue and I have already agreed on that.

MarceloSalazar commented on docs/tools/tools_intro.md:
```diff
@@ -39,7 +32,6 @@ You can build Mbed OS can be built with various toolchains. The currently suppor
   - A paid version is also included in [Keil MDK 5.29](http://www2.keil.com/mdk5/529).
-- [Arm Compiler 5.06 update 6 (to be deprecated in the future)](https://developer.arm.com/products/software-development-tools/compilers/arm-compiler-5/downloads).
 - [GNU Arm Embedded version 9 (9-2019-q4-major)](https://developer.arm.com/open-source/gnu-toolchain/gnu-rm/downloads).
 - [IAR Embedded Workbench 8.32.1](https://iar.com/mbed).
```
IAR is not supported in Mbed OS 6.0 and should be removed.

iriark01 committed:
Another go at removing tooling

Tools questions (talking with Hugues about them today)

1. mbed_targets.md - odd bit & mentions IAR - can we really force people not to use it when working on their own targets?
2. cli-compile.md - removing the list of values will mean people have to click to a different page just for that. It breaks the flow.
3. light-control.md - can I just replace make_iar and iar in the commands? 
4. porting_reqs.md - not sure I want to remove the IAR mention, as I’m not sure we can force people to change tools when they port.
5. flash.md - another porting page
6. bootstrap.md - another porting page
7. Spm.md - another porting page
8. exporter.md - less clear cut, since we can say we don’t support exporting to it anymore
9. export_to_third_party.md - again, not sure I want to remove IAR as it’s up to other people.
10. Somewhere, Hugues wanted to add something about microlibs or small c library or something, right?

iriark01 committed:
Another go at removing tooling

Tools questions (talking with Hugues about them today)

1. mbed_targets.md - odd bit & mentions IAR - can we really force people not to use it when working on their own targets?
2. cli-compile.md - removing the list of values will mean people have to click to a different page just for that. It breaks the flow.
3. light-control.md - can I just replace make_iar and iar in the commands? 
4. porting_reqs.md - not sure I want to remove the IAR mention, as I’m not sure we can force people to change tools when they port.
5. flash.md - another porting page
6. bootstrap.md - another porting page
7. Spm.md - another porting page
8. exporter.md - less clear cut, since we can say we don’t support exporting to it anymore
9. export_to_third_party.md - again, not sure I want to remove IAR as it’s up to other people.
10. Somewhere, Hugues wanted to add something about microlibs or small c library or something, right?

Link changes will be handled once this is merged

iriark01 committed:
Merge remote-tracking branch 'origin/updating-tools' into updating-tools

iriark01 committed:
Pushing final removals

Next steps will be fixing the merge conflicts

iriark01 committed:
Update docs/tools/toolchains/export_to_third_party.md

iriark01 committed:
Resolving some conflicts

iriark01 committed:
Resolving some conflicts

iriark01 committed:
Resolving conflicts

iriark01 committed:
Resolving conflicts

iriark01 committed:
Resolving conflicts

iriark01 committed:
Resolving conflicts

iriark01 committed:
Merge branch 'development' into updating-tools

iriark01 committed:
Last fixes

iriark01 committed:
Merge remote-tracking branch 'origin/updating-tools' into updating-tools
